### PR TITLE
BEMXJST: Fix for #202 applyCtx brokes this.position

### PR DESCRIPTION
**Pull request by miripiruni**
Fix for #202

### CODE REVIEW

File: test/bemcontext-position-test.js [hunk 1]
```diff
@@ -115,4 +115,21 @@ describe('BEMContext this.position', function() {
     { block: 'a1', content: { block: 'a2', content: { block: 'a3' } } },
     '<div class="a1 1"><div class="a2 1"><div class="a3 1"></div></div></div>');
   });
+
+  it('should proper calc position with applyCtx()', function() {
+    test(function() {
+      block('a')(
+        cls()(function() { return this.position; }),
+        match(function() { return !this._flag; })
```
Comment (Yeti-or) [inline test/bemcontext-position-test.js]: Why not:

**Comment by zxqfox** [comment]
Agree with @Yeti-or 

The rest LGTM

**Comment by miripiruni** [comment]
Updated.

File: lib/bemxjst/index.js [hunk 2]
```diff
@@ -52,10 +52,14 @@ BEMXJST.prototype.compile = function compile(code) {
...
 
+  function applyNewCtx() {
+    return self.run(self.context.ctx);
+  }
+
   function applyCtxWrap(ctx, changes) {
     // Fast case
     if (!changes)
-      return self.local({ ctx: ctx }, applyCtx);
+      return self.local({ ctx: ctx }, applyNewCtx);
```
Comment (zxqfox) [inline lib/bemxjst/index.js]: Can we add case with changes?

File: lib/bemxjst/index.js [hunk 3]
```diff
@@ -52,10 +52,14 @@ BEMXJST.prototype.compile = function compile(code) {
     return self._run(self.context.ctx);
```
Comment (miripiruni) [inline lib/bemxjst/index.js]: Let’s try to save position here.

**Comment by miripiruni** [comment]
Check position with `wrap` & `replace`.

**Pull request closed without merging**